File: Week_01/java/src/_66_PlusOne.java
/**
 * 第一遍：2020/08/14周五 ✅
 * 第二遍：2020/05/22周五
 * 第三遍：2020/05/25周一
 * 第四遍：2020/08/13周四
 * 第五遍：2020/08/14周五
 */
class _66_PlusOne {
    public int[] plusOne(int[] digits) {
        for (int i = digits.length - 1; i >= 0; i--) {
            digits[i]++;
            digits[i] = digits[i] % 10;
            if (digits[i] != 0) return digits;
        }
        digits = new int[digits.length + 1];
        digits[0] = 1;
        return digits;
    }
}

File: Week_01/kotlin/src/_66_PlusOne.kt
fun plusOne(digits: IntArray): IntArray {
    val n = digits.size
    for (i in n - 1 downTo 0) {
        digits[i]++
        digits[i] = digits[i] % 10
        if (digits[i] != 0) return digits
    }
    return IntArray(n + 1) { if (it == 0) 1 else 0 }
}